### ISSUE TITLE
Pipeline fails on fresh jx boot install using self hosted gitlab on AWS

### ISSUE THREAD

**Issue by orinbou** [jenkins-x/jx#6951]
### Summary 
The pipeline for the dev environment fails on a fresh jx boot install using gitlab because the error of my gitlab access.(My gitlab is self hosted on AWS)

### Steps to reproduce the behavior
1. Run jx boot with a similar jx-requirements.yml:
```
cluster:
  clusterName: eks-suna
  environmentGitOwner: my.name
  gitKind: gitlab
  gitName: gl
  gitServer: https://gitlab.xxx-yyy.net
  provider: eks
gitops: true
environments:
- key: dev
- key: staging
- key: production
ingress:
  domain: ""
  externalDNS: false
  tls:
    email: ""
    enabled: false
    production: false
kaniko: true
secretStorage: local
storage:
  logs:
    enabled: false
    url: ""
  reports:
    enabled: false
    url: ""
  repository:
    enabled: false
    url: ""
versionStream:
  ref: "master"
  url: https://github.com/jenkins-x/jenkins-x-versions.git
webhook: lighthouse
```
2. Create quickstart sample application:
```
jx create quickstart
? select the quickstart you wish to create spring-boot-http-gradle
Using Git provider gitlab.xxx-yyy.net at https://gitlab.xxx-yyy.net
? Do you wish to use my.nameas the Git user name? Yes
? Who should be the owner of the repository? my.name
? Enter the new repository name:  jx3-spring-boot-http-gradle
Creating repository my.name/jx3-spring-boot-http-gradle
Generated quickstart at /home/ec2-user/environment/jx3/jx3-spring-boot-http-gradle
Created project at /home/ec2-user/environment/jx3/jx3-spring-boot-http-gradle
The directory /home/ec2-user/environment/jx3/jx3-spring-boot-http-gradle is not yet using git
? Would you like to initialise git now? Yes
? Commit message:  Initial import

Git repository created

Watch pipeline activity via:    jx get activity -f jx3-spring-boot-http-gradle -w
Browse the pipeline log via:    jx get build logs my.name/jx3-spring-boot-http-gradle/master
You can list the pipelines via: jx get pipelines
When the pipeline is complete:  jx get applications

For more help on available commands see: https://jenkins-x.io/developing/browsing/

Note that your first pipeline may take a few minutes to start while the necessary images get downloaded!
``` 

### Expected behavior
The above pipeline should complete successfully.

### Actual behavior
The pipeline fails with the following logs:
```
jx get build log my-name/jx3-spring-boot-http-gradle/master
? Which build do you want to view the logs of?:  my-name/jx3-spring-boot-http-gradle/master #1
Build logs for my-name/jx3-spring-boot-http-gradle/master #1

Showing logs for build my-name/jx3-spring-boot-http-gradle/master #1 stage meta-pipeline and container step-credential-initializer-vwgg6
{"level":"warn","ts":1584796156.6932092,"logger":"fallback-logger","caller":"logging/config.go:69","msg":"Fetch GitHub commit ID from kodata failed: \"ref: refs/heads/0.8.0-jx-support-backwards-incompats\" is not a valid GitHub commit ID"}
{"level":"info","ts":1584796156.6935039,"logger":"fallback-logger","caller":"creds-init/main.go:40","msg":"Credentials initialized."}

Showing logs for build my-name/jx3-spring-boot-http-gradle/master #1 stage meta-pipeline and container step-working-dir-initializer-qghqc
{"level":"warn","ts":1584796157.131295,"logger":"fallback-logger","caller":"logging/config.go:69","msg":"Fetch GitHub commit ID from kodata failed: open /var/run/ko/HEAD: no such file or directory"}
{"level":"info","ts":1584796157.1336625,"logger":"fallback-logger","caller":"bash/main.go:64","msg":"Successfully executed command \"sh -c mkdir -p /workspace/source\"; output "}

Showing logs for build my-name/jx3-spring-boot-http-gradle/master #1 stage meta-pipeline and container step-place-tools

Showing logs for build my-name/jx3-spring-boot-http-gradle/master #1 stage meta-pipeline and container step-git-source-meta-my-name-jx3-spring-r55b9-6xpz5
{"level":"warn","ts":1584796161.2747355,"logger":"fallback-logger","caller":"logging/config.go:69","msg":"Fetch GitHub commit ID from kodata failed: \"ref: refs/heads/0.8.0-jx-support-backwards-incompats\" is not a valid GitHub commit ID"}
{"level":"error","ts":1584796376.9099808,"logger":"fallback-logger","caller":"git/git.go:40","msg":"Error running git [fetch --depth=1 --recurse-submodules=yes origin master]: exit status 128\nfatal: unable to access 'https://gitlab.xxx-yyy.net/my.name/jx3-spring-boot-http-gradle.git/': Failed to connect to gitlab.xxx-yyy.net port 443: Operation timed out\n","stacktrace":"github.com/tektoncd/pipeline/pkg/git.run\n\t/Users/abayer/go/src/github.com/tektoncd/pipeline/pkg/git/git.go:40\ngithub.com/tektoncd/pipeline/pkg/git.Fetch\n\t/Users/abayer/go/src/github.com/tektoncd/pipeline/pkg/git/git.go:91\nmain.main\n\t/Users/abayer/go/src/github.com/tektoncd/pipeline/cmd/git-init/main.go:39\nruntime.main\n\t/usr/local/Cellar/go/1.12.4/libexec/src/runtime/proc.go:200"}
{"level":"error","ts":1584796592.2456024,"logger":"fallback-logger","caller":"git/git.go:40","msg":"Error running git [pull --recurse-submodules=yes origin]: exit status 1\nfatal: unable to access 'https://gitlab.xxx-yyy.net/my.name/jx3-spring-boot-http-gradle.git/': Failed to connect to gitlab.xxx-yyy.net port 443: Operation timed out\n","stacktrace":"github.com/tektoncd/pipeline/pkg/git.run\n\t/Users/abayer/go/src/github.com/tektoncd/pipeline/pkg/git/git.go:40\ngithub.com/tektoncd/pipeline/pkg/git.Fetch\n\t/Users/abayer/go/src/github.com/tektoncd/pipeline/pkg/git/git.go:94\nmain.main\n\t/Users/abayer/go/src/github.com/tektoncd/pipeline/cmd/git-init/main.go:39\nruntime.main\n\t/usr/local/Cellar/go/1.12.4/libexec/src/runtime/proc.go:200"}
{"level":"warn","ts":1584796592.2456717,"logger":"fallback-logger","caller":"git/git.go:95","msg":"Failed to pull origin : exit status 1"}
{"level":"error","ts":1584796592.2470903,"logger":"fallback-logger","caller":"git/git.go:40","msg":"Error running git [checkout master]: exit status 1\nerror: pathspec 'master' did not match any file(s) known to git\n","stacktrace":"github.com/tektoncd/pipeline/pkg/git.run\n\t/Users/abayer/go/src/github.com/tektoncd/pipeline/pkg/git/git.go:40\ngithub.com/tektoncd/pipeline/pkg/git.Fetch\n\t/Users/abayer/go/src/github.com/tektoncd/pipeline/pkg/git/git.go:97\nmain.main\n\t/Users/abayer/go/src/github.com/tektoncd/pipeline/cmd/git-init/main.go:39\nruntime.main\n\t/usr/local/Cellar/go/1.12.4/libexec/src/runtime/proc.go:200"}
{"level":"fatal","ts":1584796592.2471752,"logger":"fallback-logger","caller":"git-init/main.go:40","msg":"Error fetching git repository: exit status 1","stacktrace":"main.main\n\t/Users/abayer/go/src/github.com/tektoncd/pipeline/cmd/git-init/main.go:40\nruntime.main\n\t/usr/local/Cellar/go/1.12.4/libexec/src/runtime/proc.go:200"}

Pipeline failed on stage 'meta-pipeline' : container 'step-git-source-meta-my-name-jx3-spring-r55b9-6xpz5'. The execution of the pipeline has stopped.
```
In this environment, I can connect to gitlab and perform manual operations (clone, pull, push, etc.). Therefore, it is difficult for me to understand why jx encounters a connection error to gitlab during pipeline processing.

### Jx version

The output of `jx version` is:
```
NAME               VERSION
jx                 2.0.1249
Kubernetes cluster v1.14.9-eks-502bfb
kubectl            v1.14.7-eks-1861c5
helm client        2.12.2
git                2.25.1
Operating System   Unknown Linux distribution Linux version 4.14.165-103.209.amzn1.x86_64 (mockbuild@koji-pdx-corp-builder-60005) (gcc version 7.2.1 20170915 (Red Hat 7.2.1-2) (GCC)) #1 SMP Sun Feb 9 00:23:26 UTC 2020

verifying packages
```

### Jenkins type

- [x] Serverless Jenkins X Pipelines (Tekton +  ~~Prow~~) <- lighthouse
- [ ] Classic Jenkins

### Kubernetes cluster

AWS-EKS on EC2  
Kubernetes ver 1.14  
Create by CloudFormation

### Operating system / Environment

AWS Cloud9 on EC2(Amazon Linux)  
```
/etc/os-release:NAME="Amazon Linux AMI"
/etc/os-release:VERSION="2018.03"
/etc/os-release:ID="amzn"
/etc/os-release:ID_LIKE="rhel fedora"
/etc/os-release:VERSION_ID="2018.03"
/etc/os-release:PRETTY_NAME="Amazon Linux AMI 2018.03"
/etc/os-release:ANSI_COLOR="0;33"
/etc/os-release:CPE_NAME="cpe:/o:amazon:linux:2018.03:ga"
/etc/os-release:HOME_URL="http://aws.amazon.com/amazon-linux-ami/"
/etc/system-release:Amazon Linux AMI release 2018.03
```

**Comment by deanesmith** [comment]
Hi @orinbou, as you pointed out, it is connectivity error.  This error is coming from a pod in your cluster, your access rights to your Gitlab instance are the issue - `unable to access 'https://gitlab.xxx-yyy.net/my.name/jx3-spring-boot-http-gradle.git/': Failed to connect to gitlab.xxx-yyy.net port 443`.  Do any other services successfully establish a connection?

**Comment by orinbou** [comment]
@deanesmith Thanks for your reply.
I noticed my mistake by your reply.
It is in the EKS worker nodes that the build runs on JenkinsX.
However, I did not give the EKS worker nodes proper permissions on AWS.
Therefor, I added pammisions to worker node of EC2, and I retried. 
As a result, the build pipeline was executed.
Sorry. And Thank you very much for your precious time.
I will want to close the ticket.
